### DISCUSSION TITLE
Need to understand routing

### DISCUSSION THREAD

**Discussion by Gotiap** [oqtane/oqtane.framework#1193] [General]
how routing works like the login page and other pages I mean how it mapped to **/login** URL with **Modules/Admin/Login/Index**.
 
And also I need to know, As you can see attached screed shot, multiple sub-modules are working as expected but now I have one more requirement like In CSO.Module1 I have **Test/Edit.razor** file I want to route like this **http://localhost:44357/mypage/*/29/Test/Edit** instead of  **http://localhost:44357/mypage/*/29/Edit**, As of now  **http://localhost:44357/mypage/*/29/Edit**  URL working as expected. 

![0918B4A81DFC4EC9A4A61F9335BA74EF](https://user-images.githubusercontent.com/28126997/111935195-fa38bc00-8ae8-11eb-90be-707abbab64e3.png)

Thank you for your time

**Comment by sbwalker** [comment]
Is there a reason you cannot simply rename the component to TestEdit.razor? Then your module Url would be http://localhost:44357/mypage/*/29/TestEdit. Note that there is no relationship between routes in Oqtane and the way files are organized in the underlying file system - they are completely independent concepts. Internally, routing is actually based on type names

**Reply by Gotiap** [reply]
I want to create some sub-designs in a specific folder. So I want to separate by folders.

So actually how this map to pages?

**Reply by sbwalker** [reply]
Oqtane is a modular framework... "pages" are a virtual configuration concept, not a physical concept. I would suggest you watch some of the introductory videos to understand the approach: https://www.youtube.com/watch?v=JPfUZPlRRCE&list=PLYhXmd7yV0elLNLfQwZBUlM7ZSMYPTZ_f

**Reply by Gotiap** [reply]
Thank you.
